reconcile/terraform_aws_route53.py

```diff
@@ -37,8 +37,23 @@ def build_desired_state(
         account = zone["account"]
         account_name = account["name"]
 
-        zone_name = zone["name"]
-        zone_values = {"name": zone_name, "account_name": account_name, "records": []}
+        # optionally decouple the name of the DNS file from the domain
+        # it is creating records for to allow split view DNS
+        # https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/hosted-zone-private-considerations.html#hosted-zone-private-considerations-split-view-dns
+        domain_name = zone.get("domain_name")
+        if domain_name:
+            zone_name = domain_name
+            zone_id = zone["name"]
+        else:
+            zone_name = zone["name"]
+            zone_id = zone["name"]
```

Comment:
To avoid continued confusion with the lack of actual zone-ids being used, could we use variable name `resource_name` instead of `zone_id`?

Reply:
That sounds good yeah, I agree that `id` is being used wayyyy too much here